paperqa/contrib/zotero.py

```diff
@@ -44,8 +45,22 @@ def __init__(
         library_id: Optional[str] = None,
         api_key: Optional[str] = None,
         storage: Optional[StrPath] = None,
+        zotero_storage: Optional[Union[StrPath,bool]] = "~/Zotero/storage/",
```

Comment:
```suggestion
        zotero_storage: Optional[Union[StrPath,bool]] = None,
```

And initialize it later on, based on what operating system it is.

Also, should use `pathlib.Path.home() / "Zotero" / "storage"` for safety

Reply:
Ok so we could do:

```
Optional[Union[StrPath,bool]] = None

StrPath: try to use the user-provided path, raise FileNotFoundError if it doesn't exist
None: try to use pathlib.Path.home() / "Zotero" / "storage", if it doesn't exist, set zotero_storage to None
False: never use zotero_storage
```

But what about

```
Union[StrPath,bool] = True

StrPath: try to use the user-provided path, raise FileNotFoundError if it doesn't exist
True: try to use pathlib.Path.home() / "Zotero" / "storage", if it doesn't exist, set zotero_storage to None
False: never use zotero_storage
```

I think default True makes it more obvious that the functionality is engaged by default. WYT?

Reply:
Also out of curiosity what's the diff btw `pathlib.Path.home() / "Zotero" / "storage"` and `Path("~/Zotero/storage/").expanduser()`?

Reply:
> <Pathlib question>

It's just a more robust practice: On Windows, the "\" key is used instead of "/". So Pathlib's `/` operator will do the correct thing on each operating system. 

Also sometimes the `"~"` is not expanded correctly. Better to rely on Pathlib.

Reply:
Being explicit is often best so I would recommend one parameter for specifying the location, and another for specifying whether you should use it or not. When variables can take on a bunch of different types it can make code confusing. (I edited my original comment with this suggestion)